File: 555-Parsing/61-SLK/slk-example/cs/ClcLog.cs

using System ;

class ClcLog {

public
void
trace ( string  message )
{
    Console.WriteLine ( message );
}

public
void
trace_depth ( string message,
              int    depth )
{
    int     i;

    for ( i = 0;  i < depth;  ++i ) {
        Console.WriteLine ( "    " );
    }
    Console.WriteLine ( message );
}

public
void
trace_production ( short  production_number )
{
    Console.WriteLine ( ClcString.GetProductionName ( production_number ) );
    Console.WriteLine ( "\n" );
}

public
void
trace_action ( short  action_number )
{
    Console.WriteLine ( "\n" );
    Console.WriteLine ( ClcString.GetSymbolName ( action_number ) );
    Console.WriteLine ( "\n\n" );
}

};

File: 555-Parsing/61-SLK/slk-example/java/ClcLog.java

class ClcLog {

public
void
trace ( String  message )
{
    System.out.print ( message );
}

public
void
trace_depth ( String message,
              int    depth )
{
    int     i;

    for ( i = 0;  i < depth;  ++i ) {
        System.out.print ( "    " );
    }
    System.out.print ( message );
}

public
void
trace_production ( short  production_number )
{
    System.out.print ( ClcString.GetProductionName ( production_number ) );
    System.out.print ( "\n" );
}

public
void
trace_action ( short  action_number )
{
    System.out.print ( "\n" );
    System.out.print ( ClcString.GetSymbolName ( action_number ) );
    System.out.print ( "\n\n" );
}

};
